framework/BaseYii.php

```diff
@@ -362,7 +364,7 @@ public static function getLogger()
             return self::$_logger;
         }
 
-        return self::$_logger = static::createObject('yii\log\Logger');
+        throw new InvalidConfigException('No logger without application');
```

Comment:
I think this exception will be popular so we should explicitly explain what's going on.

How about `Logger component must be initialized before using logging actions. Add "log" component to the "bootstrap" section of the configuration on the first position.`?